branchpro/apps/_simulation.py

```diff
@@ -123,7 +152,30 @@ def add_collapsed_text(self, text, title='More details...'):
             ])
         self.app.layout.children[0].children[-1].children.append(collapse)
 
-    def add_data(self, df, time_label='Time', inc_label='Incidence Number'):
+    def parse_contents(self, contents, filename):
+        content_type, content_string = contents.split(',')
+
+        decoded = base64.b64decode(content_string)
+        try:
+            if 'csv' in filename or 'txt' in filename:
+                # Assume that the user uploaded a CSV or TXT file
+                df = pd.read_csv(
+                    io.StringIO(decoded.decode('utf-8')))
+            elif 'xls' in filename:
```

Comment:
I don't mind if we force the user to use csv/txt. But if we do support excel imports, this check should be slightly more general. According to pandas the read_excel method can take xls, xlsx, xlsm, xlsb, odf, ods and odt. If we provide this support we should probably add openpyxl, xlrd, and odfpy to the dependencies.